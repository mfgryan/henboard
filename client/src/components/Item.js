// react dep
import React from "react";
import { Row, Col } from "react-bootstrap";

// component dep
import InfoItem from "./containers/InfoItem.js";

const Item = ({
    draggable,
    item,
    children,
    dragItem,
    itemClicked,
    removeItem
}) => {
    return (
        <Row
            className="Item"
            draggable={draggable}
            onDragStart={event => dragItem(event, item)}>
            <Col md={10}>
                <p onClick={event => itemClicked(event, item)}>
                    {item.name}
                </p>
            </Col>
            <Col md={2}>
                <p onClick={() => removeItem(item)}>
                    {item.check ? "âœ”" : "X"}
                </p>
            </Col>
            {children}
        </Row>
    );
};

export default Item;
