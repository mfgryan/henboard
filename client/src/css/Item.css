.removeItem {
    visibility: hidden;
}

.Item:hover .removeItem {
    visibility: visible;
}
