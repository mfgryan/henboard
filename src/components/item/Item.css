.itemval{
    font-size: 1.5em; 
}
.info{
    font-size: 1.3em;
    font-weight: bold;
}
