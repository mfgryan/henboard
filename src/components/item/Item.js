import React, { Component } from "react";
import { Button } from "react-bootstrap";
import "./Item.css";
import Dialog from "../dialog/Dialog";
import Swimlane from "../swimlane/Swimlane";

class Item extends React.Component {
    constructor(props){
        super(props);     
        this.state = {
            isOpen: false 
        }
        this.openInfo = this.openInfo.bind(this);
        this.closeInfo = this.closeInfo.bind(this);
        this.drag = this.drag.bind(this);
    }
    openInfo(){
        this.setState({isOpen: true}); 
    }
    closeInfo(){
        this.setState({isOpen: false}); 
    }
    drag(e){
        const item = {val: this.props.val, info: this.props.info};
        e.dataTransfer.setData("item", JSON.stringify(item));
        this.props.setDragLane();
    }
    render(){
        const body = <Swimlane name={this.props.val} items={this.props.info}/>;
        const footer = <Button onClick={this.closeInfo}>Close</Button>;
        return (
            <p draggable="true" onDragStart={this.drag} >
                {this.props.info &&  <span className="alignleft blue info" onClick={this.openInfo}> i </span>}
                <span className="aligncenter itemval">{this.props.val}</span>
                {!this.props.hideRemove &&  <span className="alignright red" id={this.props.val} onClick={() => this.props.removeItem(this.props.val)}>X</span>}
                {this.props.info && <Dialog body={body} footer={footer} isOpen={this.state.isOpen}/>}
            </p> 
        ); 
    }
}
export default Item;
