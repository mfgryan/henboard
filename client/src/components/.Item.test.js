import React from "react";
import Item from "./Item";
import renderer from "react-test-renderer";

it('Item renders correctly', () => {
    let item = {
        project: "henboard",  
        name: "test",  
        column: "todo",  
        value: "foo",  
        week: "01/01/01",  
        openInfo: false,  
        addItem: false
    }
    const tree = renderer.create(
        <Item draggable item={item} dragItem={() => {}} itemClicked={() => {}} removeItem={() => {}}>
            test
        </Item>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
