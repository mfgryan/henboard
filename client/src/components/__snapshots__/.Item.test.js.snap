exports[`test Item renders correctly 1`] = `
<div
  className="Item row"
  draggable={true}
  onDragStart={[Function]}>
  <div
    className="col-md-10">
    <p
      onClick={[Function]}>
      test
    </p>
  </div>
  <div
    className="col-md-2">
    <p
      onClick={[Function]}>
      X
    </p>
  </div>
  test
</div>
`;
